### PR TITLE
Simplify the way we explain how to test cookies in controller

### PR DESCRIPTION
**Pull request by benoittgt**
The previous documentation was old and was not clear enough to help
developers to understand properly how to test cookies inside a
controller test. Since then it is much easier to test cookies if you
stick to `cookies` and don't use `response.cookies`.

You can use response.cookies when you don't change cookies inside your
test.

```ruby
def show
  cookies["user_name"] = nil
  head 200
end

RSpec.describe SignOutsController, type: :request do
  describe 'GET /signout' do
    it "clear cookie value" do
      get "/sign_out"

      expect(response.cookies["user_name"]).to eq(nil)
    end
  end
end
```

Use `cookies` to set and expect when you set cookies inside your test

```ruby
def show
  cookies.delete("user_name")
  head 200
end

RSpec.describe SignOutsController, type: :request do
  describe 'GET /signout' do
    it "clear cookie value" do
      cookies["user_name"] = "Sam"

      get "/sign_out"

      expect(cookies["user_name"]).to eq("")
    end
  end
end
```

This is the preferred way.

Related:
- https://github.com/rspec/rspec-rails/issues/1993

### CODE REVIEW

**Comment by benoittgt** [comment]
Can we stick rubocop version...? It is failing way too often.

File: features/controller_specs/cookies.feature [hunk 1]
```diff
@@ -0,0 +1,36 @@
+Feature: Cookies
+
+  There is different ways to test cookies in controller's tests be we recommend
+  to use `cookies` as below.
```
Comment (JonRowe) [inline features/controller_specs/cookies.feature]: I think this needs rewording, theres a few grammatical issues. How about:
```
There are different ways to make assertions on cookies from controller specs, but we recommend using the `cookies` method as set out below.
```

File: features/controller_specs/cookies.feature [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  You can use strings or symbols to fetch or set your cookies because `cookies`
+  support indifferent access.
```
Comment (JonRowe) [inline features/controller_specs/cookies.feature]: Swap out '`cookies` support' for 'the `cookies' method supports'

File: features/controller_specs/cookies.feature (hunk 1 above)
Comment (benoittgt) [reply features/controller_specs/cookies.feature]: Thanks. It is much better with your proposal. Changed with https://github.com/rspec/rspec-rails/pull/2127/commits/54104d8ea3ce6c4f75be17518f34593799930371

File: features/controller_specs/cookies.feature (hunk 2 above)
Comment (benoittgt) [reply features/controller_specs/cookies.feature]: Thanks. Done with https://github.com/rspec/rspec-rails/pull/2127/commits/54104d8ea3ce6c4f75be17518f34593799930371

**Comment by JonRowe** [comment]
LGTM, merge on green!

**Pull request merged**